docs/manual/debugging.md

```diff
@@ -0,0 +1,133 @@
+✔ Cluster is available
+✔ Cluster is fully replicated
+✖ Cluster is not reconciled
+✖ ProcessGroup: stateless-5 has the following condition: MissingProcesses since 2021-04-27 02:57:25 +0100 BST
+✖ ProcessGroup: storage-2 has the following condition: MissingProcesses since 2021-03-25 07:31:08 +0000 GMT
+✖ ProcessGroup: storage-2 has the following condition: PodFailing since 2021-03-25 18:23:22 +0000 GMT
+✖ ProcessGroup: storage-2 has the following condition: IncorrectConfigMap since 2021-03-26 19:20:56 +0000 GMT
+✖ ProcessGroup: transaction-5 has the following condition: MissingProcesses since 2021-04-27 02:57:24 +0100 BST
+✖ Pod default/example-cluster-storage-2 has unexpected Phase Pending with Reason:
+✖ Pod default/example-cluster-storage-2 has an unready container: foundationdb
+✖ Pod default/example-cluster-storage-2 has an unready container: foundationdb-kubernetes-sidecar
+✖ Pod default/example-cluster-storage-2 has an unready container: trace-log-forwarder
+Error:
+found issues for cluster example-cluster. Please check them
+```
+
+The plugin can also resolve these issues automatically:
+
+```
+$ kubectl fdb analyze example-cluster --auto-fix
+Checking cluster: default/example-cluster
+✔ Cluster is available
+✔ Cluster is fully replicated
+✖ Cluster is not reconciled
+✖ ProcessGroup: stateless-5 has the following condition: MissingProcesses since 2021-04-27 02:57:25 +0100 BST
+✖ ProcessGroup: storage-2 has the following condition: MissingProcesses since 2021-03-25 07:31:08 +0000 GMT
+✖ ProcessGroup: storage-2 has the following condition: PodFailing since 2021-03-25 18:23:22 +0000 GMT
+✖ ProcessGroup: storage-2 has the following condition: IncorrectConfigMap since 2021-03-26 19:20:56 +0000 GMT
+✖ ProcessGroup: transaction-5 has the following condition: MissingProcesses since 2021-04-27 02:57:24 +0100 BST
+✖ Pod default/example-cluster-storage-2 has unexpected Phase Pending with Reason:
+✖ Pod default/example-cluster-storage-2 has an unready container: foundationdb
+✖ Pod default/example-cluster-storage-2 has an unready container: foundationdb-kubernetes-sidecar
+✖ Pod default/example-cluster-storage-2 has an unready container: trace-log-forwarder
+Replace instances [stateless-5 storage-2 transaction-5] in cluster default/example-cluster [y/n]: y
+Error:
+found issues for cluster example-cluster. Please check them
+```
+
+When using this feature, read carefully what the plugin wants to do and only confirm the dialog when you are sure that you want to do these actions.
+
+## Pods stuck in Pending
+
+If you have Pods that are failing to launch, because they are stuck in either a pending or terminating state, you can address that by replacing the failing instance. You can do that using a [plugin command](#replacing-pods-with-the-kubectl-plugin).
+
+## Replacing Pods with the kubectl plugin
+
+Let's assume we are working with the cluster `example-cluster`, and the pod `example-cluster-storage-1` is failing to launch.
```

Comment:
The assumption here is that the Pod was at least once up and running and got an IP address otherwise the `failing to launch` will result in a process group without an IP address and the replacement with exclusion will fail.